tpm2/tpm2.go

```diff
@@ -1235,30 +1235,35 @@ func NVDefineSpaceEx(rw io.ReadWriter, owner tpmutil.Handle, ownerAuth, authVal
 	return err
 }
 
-func encodeWriteNV(owner, handle tpmutil.Handle, authString string, data tpmutil.U16Bytes, offset uint16) ([]byte, error) {
-	auth, err := encodeAuthArea(AuthCommand{Session: HandlePasswordSession, Attributes: AttrContinueSession, Auth: []byte(authString)})
+// NVWrite writes data into the TPM's NV storage.
+func NVWrite(rw io.ReadWriter, owner, handle tpmutil.Handle, authString string, data tpmutil.U16Bytes, offset uint16) error {
```

Comment:
nit: can you use the TPM spec names for these handles: `authHandle` and `nvIndex`? Calling the first handle `owner` (which is a problem that predates this PR) is unnecessarily prescriptive. The NV index itself could be the source of the authorization value, e.g., if the NV index has some authorization policy on it.